src/Core/Managed/Shared/Channel/InMemoryTransmitter.cs

```diff
@@ -133,18 +133,22 @@ private void DequeueAndSend(TimeSpan timeout)
         /// <summary>
         /// Serializes a list of telemetry items and sends them.
         /// </summary>
-        private async Task Send(IEnumerable<ITelemetry> telemetryItems, TimeSpan timeout)
+        private Task Send(IEnumerable<ITelemetry> telemetryItems, TimeSpan timeout)
         {
             if (telemetryItems == null || !telemetryItems.Any())
             {
                 CoreEventSource.Log.LogVerbose("No Telemetry Items passed to Enqueue");
-                return;
+#if NET40
+                return TaskEx.FromResult<object>(null);
+#else
+                return Task.FromResult<object>(null);
+#endif
             }
 
             byte[] data = JsonSerializer.Serialize(telemetryItems);
             var transmission = new Transmission(this.endpointAddress, data, "application/x-json-stream", JsonSerializer.CompressionType, timeout);
 
-            await transmission.SendAsync().ConfigureAwait(false);
+            return transmission.SendAsync();
```

Comment:
We should not change the behavior.

---
In reply to: [73466495](https://github.com/Microsoft/ApplicationInsights-dotnet/pull/267#discussion_r73466495) [](ancestors = 73466495)